cmake/modules/SwiftSupport.cmake

```diff
@@ -0,0 +1,94 @@
+
+# Returns the current architecture name in a variable
+#
+# Usage:
+#   get_swift_host_arch(result_var_name)
+#
+# If the current architecture is supported by Swift, sets ${result_var_name}
+# with the sanitized host architecture name derived from CMAKE_SYSTEM_PROCESSOR.
+function(get_swift_host_arch result_var_name)
```

Comment:
We can just ask the compiler for the module triple instead and remove this tower of architectures.